libsupport/include/katana/RDGVersion.h

```diff
@@ -0,0 +1,60 @@
+#ifndef KATANA_LIBSUPPORT_KATANA_RDGVERSION_H_
+#define KATANA_LIBSUPPORT_KATANA_RDGVERSION_H_
+
+#include <cstring>
+#include <iostream>
+#include <iterator>
+#include <vector>
+
+#include <fmt/format.h>
+#include <nlohmann/json.hpp>
+
+#include "katana/Random.h"
+#include "katana/config.h"
+
+namespace katana {
+
+const uint64_t kRDGVersionMaxID = (1 << 30);
+const uint64_t kRDGVersionIDLength = (256);
+const int      kRDGVersionPaddingLength = (10);
+
+struct KATANA_EXPORT RDGVersion {
+  // A vectorized version in the form of num:id
+  // The last one has an empty branch "".
+  std::vector<uint64_t> numbers_{0};
+  std::vector<std::string> branches_{"."};
+
+  RDGVersion(const RDGVersion& in) = default;
+  RDGVersion& operator=(const RDGVersion& in) = default;
+
+  RDGVersion(
+      const std::vector<uint64_t>& nums,
+      const std::vector<std::string>& branches);
+  RDGVersion(const std::string& str);
+  explicit RDGVersion(uint64_t num = 0);
+
+  // Accessors
+  std::string ToPathName() const;
+  std::string ToString() const;
+  uint64_t LeafNumber() const;
+  bool ShareBranch(const RDGVersion& in) const;
+  bool NullNumber() const;
+  bool NullBranch() const;
+  bool IsNull() const;
```

Comment:
Null is really really dangerous. I would STRONGLY suggest you remove the idea of a null version and use `optional<RDGVersion>` anywhere the version may or may not be provided.

Reply:
As I said above, I think this should be `IsZero`. Then this all makes sense. Null generally refers to the lack of something, not the first of something.